.github/workflows/scripts/slack_gem_notifications/notifications_methods.rb

```diff
@@ -0,0 +1,48 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'time'
+require 'httparty'
+
+def check_for_updates(watched_gems)
+  return if gem_list_empty?(watched_gems)
+  watched_gems.each do |gem_name|
+    is_valid_gem?(gem_name) ? gem_info = is_valid_gem?(gem_name) : return
+    versions = gem_versions(gem_info)
+    send_bot(gem_name, versions) if gem_updated?(versions)
+  end
+end
+
+def gem_list_empty?(watched_gems)
+  if watched_gems.empty?
+    raise 'Nothing to see here! watched_gems cannot be empty'
+    exit
```

Comment:
Nice use of `exit`! I wasn't familiar with that method and learned a bit more in, https://www.rubyguides.com/2019/03/ruby-exit/

It looks like you could accomplish what's on line 19 and 20 using `abort 'Nothing to see here! watched_gems cannot be empty'` if you'd like!

Reply:
`exit` when not passed any arguments will cause the Ruby application to exit with a code 0, signaling that everything worked fine. We should have been exiting here with a non 0 code to signal that there was an issue:

```ruby
exit false
# or
exit 42
```

Another nice thing about `abort` is that it automatically calls `exit false` for the desired non-zero exit code.

Great suggestion, @kaylareopelle! We should go with it!